libdrgn/dwarf_index.c

```diff
@@ -2125,23 +2199,34 @@ static struct drgn_error *index_die(struct drgn_dwarf_index *dindex,
 	}
 
 	index = die - shard->dies.data;
-	if (!append_die_entry(shard, tag, file_name_hash, module, offset)) {
+	if (!append_die_entry(shard, tag, file_name_hash, module, offset, ns->dindex)) {
 		err = &drgn_enomem;
 		goto out;
 	}
+	new_die = &shard->dies.data[shard->dies.size - 1];
 	shard->dies.data[index].next = shard->dies.size - 1;
 	err = NULL;
 out:
+	if (!err && new_die && new_die->tag == DW_TAG_namespace) {
```

Comment:
When will `new_die` be `NULL` here?

Reply:
I was worried about any corner cases causing a segfault here, but I think when `!err` this should be non-null. I'll change it back.